### ISSUE TITLE
Lint/Void autocorrect fails with CheckForMethodsWithNoSideEffects: true

### ISSUE THREAD

**Issue by tsugimoto** [rubocop/rubocop#11972]
Autocorrect for `Lint/Void` fails if `CheckForMethodsWithNoSideEffects` configuration is set to true.

--------

## Expected behavior

RuboCop does not error.

## Actual behavior

```
$ rubocop -Ad --cache=false --only Lint/Void
For ...: configuration from .../.rubocop.yml
Default configuration from .../rubocop-1.52.1/config/default.yml
Inspecting 1 file
Scanning .../main.rb
An error occurred while Lint/Void cop was inspecting .../main.rb:3:0.
undefined method `send_node' for s(:def, :merge,
  s(:args), nil):RuboCop::AST::DefNode

                        node.send_node
                            ^^^^^^^^^^
Did you mean?  send_type?
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:204:in `autocorrect_nonmutating_send'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:160:in `block in check_nonmutating'
.../rubocop-1.52.1/lib/rubocop/cop/base.rb:377:in `correct'
.../rubocop-1.52.1/lib/rubocop/cop/base.rb:181:in `add_offense'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:157:in `check_nonmutating'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:101:in `check_expression'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:90:in `block in check_begin'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:90:in `each'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:90:in `check_begin'
.../rubocop-1.52.1/lib/rubocop/cop/lint/void.rb:81:in `on_begin'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `each'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:69:in `on_begin'
.../rubocop-ast-1.28.1/lib/rubocop/ast/traversal.rb:20:in `walk'
.../rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
.../rubocop-1.52.1/lib/rubocop/cop/team.rb:156:in `investigate_partial'
.../rubocop-1.52.1/lib/rubocop/cop/team.rb:91:in `investigate'
.../rubocop-1.52.1/lib/rubocop/runner.rb:345:in `block in inspect_file'
.../rubocop-1.52.1/lib/rubocop/runner.rb:344:in `each'
.../rubocop-1.52.1/lib/rubocop/runner.rb:344:in `flat_map'
.../rubocop-1.52.1/lib/rubocop/runner.rb:344:in `inspect_file'
.../rubocop-1.52.1/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
.../rubocop-1.52.1/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
.../rubocop-1.52.1/lib/rubocop/runner.rb:314:in `loop'
.../rubocop-1.52.1/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
.../rubocop-1.52.1/lib/rubocop/runner.rb:283:in `do_inspection_loop'
.../rubocop-1.52.1/lib/rubocop/runner.rb:164:in `block in file_offenses'
.../rubocop-1.52.1/lib/rubocop/runner.rb:189:in `file_offense_cache'
.../rubocop-1.52.1/lib/rubocop/runner.rb:163:in `file_offenses'
.../rubocop-1.52.1/lib/rubocop/runner.rb:154:in `process_file'
.../rubocop-1.52.1/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
.../rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each'
.../rubocop-1.52.1/lib/rubocop/runner.rb:134:in `reduce'
.../rubocop-1.52.1/lib/rubocop/runner.rb:134:in `each_inspected_file'
.../rubocop-1.52.1/lib/rubocop/runner.rb:120:in `inspect_files'
.../rubocop-1.52.1/lib/rubocop/runner.rb:73:in `run'
.../rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
.../rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
.../rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
.../rubocop-1.52.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
.../rubocop-1.52.1/lib/rubocop/cli/command.rb:11:in `run'
.../rubocop-1.52.1/lib/rubocop/cli/environment.rb:18:in `run'
.../rubocop-1.52.1/lib/rubocop/cli.rb:118:in `run_command'
.../rubocop-1.52.1/lib/rubocop/cli.rb:125:in `execute_runners'
.../rubocop-1.52.1/lib/rubocop/cli.rb:51:in `block in run'
.../rubocop-1.52.1/lib/rubocop/cli.rb:77:in `profile_if_needed'
.../rubocop-1.52.1/lib/rubocop/cli.rb:43:in `run'
.../rubocop-1.52.1/exe/rubocop:19:in `block in <top (required)>'
.../lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
.../rubocop-1.52.1/exe/rubocop:19:in `<top (required)>'
.../bin/rubocop:25:in `load'
.../bin/rubocop:25:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/Void cop was inspecting .../main.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.28.1, running on ruby 3.1.4) [x86_64-darwin20]
Finished in 0.15327999999863096 seconds
$ 
```

## Steps to reproduce the problem

.rubocop.yml 

```yaml
AllCops:
  NewCops: enable

Lint/Void:
  CheckForMethodsWithNoSideEffects: true
```

main.rb 

```rb
# frozen_string_literal: true

def merge
end

nil
```

## RuboCop version

```
$ rubocop -V
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.28.1, running on ruby 3.1.4) [x86_64-darwin20]
$ 
```

**Comment by tsugimoto** [comment]
Thanks!
